docs/how-to/best-practices-for-ipfs-builders.md

```diff
@@ -0,0 +1,75 @@
+---
+title: "Best practices for IPFS builders"
+description: "Some IPFS features are disabled or not defaults, but we encourage their use under the right circumstances. We list them here for easy access by busy builders."
+---
+
+# Best practices for IPFS builders
+
+Some IPFS features are not enabled or set as defaults, but we encourage you to use them under the right circumstances.
+
+## When is CIDv1 the best choice
+
+There are two versions of context identifiers (CIDs): CIDv0 and CIDv1.
+
+CIDv0 is simpler but much less flexible than CIDv1. It doesn't offer the future-proof and case-insensitive addressing that CIDv1 offers. Many of the existing IPFS tools still generate CIDs in v0. You can quickly tell the difference between v0 and v1 CIDs since v0 CIDs always start with `Qm`.
+
+<!-- TODO: add examples -->
+
+Some features use CIDv1 by default:
+
+- `files` ([Mutable File System](file-systems.md#mutable-file-system-mfs))
+- `object` operations
+
+We recommend you use CIDv1 when you want:
+
+- future-proof addressing
+- case-insensitive addressing
+
+To opt in, use the `--cid-version` flag in the CLI: 
+
+```shell
+ipfs add --cid-version 1
+```
+
+For more information on content addressing and CID versions, see [Content Addressing and CIDs](https://docs.ipfs.io/concepts/content-addressing/#content-addressing-and-cids).
+
+## Enable pubsub if you need fast IPNS
+
+Pubsub is one of the available options for configuring your IPFS node. It allows you to publish and subscribe to messages on a given topic. Pubsub is an experimental feature and it should not be used in a production environment. It is disabled by default.  
+
+To use this feature, use the `--enable-pubsub-experiment` option when starting the IPFS daemon:
+
+```shell
+ipfs --enable-pubsub-experiment
+```
+
+See [IPFS pubsub](../reference/cli/#ipfs-pubsub) for more information on the available subcommands.
```

Comment:
While the `--enable-pubsub-experiment` works, we now have a proper configuration key, which provides better experience:

````suggestion
To use this feature, enable  `Ipns.UsePubsub` config flag before starting the IPFS daemon:

```shell
ipfs config --json Ipns.UsePubsub true
ipfs daemon
```

From now on, IPNS will be resolved using both DHT and pubsub. Be mindful, that this is still an experimental feature, with known limitations listed [here](https://github.com/ipfs/go-ipfs/blob/master/docs/experimental-features.md#ipns-pubsub).

See [`ipfs name pubsub --help](../reference/cli/#ipfs-name-pubsub) for more information on the available subcommands.
````